Builds/CMake/CMakeFuncs.cmake

```diff
@@ -108,3 +108,10 @@ macro(group_sources curdir)
   group_sources_in(${PROJECT_SOURCE_DIR} ${curdir})
 endmacro()
 
+macro (exclude_if_included target_)
+  if (NOT ${CMAKE_CURRENT_SOURCE_DIR} STREQUAL ${CMAKE_SOURCE_DIR})
```

Comment:
It would be nice to be able to leverage is_root_project here to avoid the duplicate code, but we can't easily given the scope of things.

When refactoring the CMake recently, did you consider bringing in cmakepp as a submodule in order to have CMake functions with return values without having to set things in PARENT_SCOPE in order to reuse code?